### ISSUE TITLE
istio.io test regression with https://github.com/istio/istio/pull/40320

### ISSUE THREAD

**Issue by ericvn** [istio/istio#40467]
### Bug Description

When the subject PRs cherry-picked arrived in the 1.15 and 1.14 branches, several istio.io tests are failing:

- https://preliminary.istio.io/latest/docs/tasks/traffic-management/request-timeouts/
- https://preliminary.istio.io/latest/docs/tasks/traffic-management/egress/egress-control/

Investigate if this is a regression.

### Version

```prose
istio.io tests which include https://github.com/istio/istio/pull/40320 or it's cherry-picks
```

**Comment by ericvn** [comment]
Possibly fixed by https://github.com/istio/istio/pull/40449 and its cherry-picks. Will test.

**Comment by ericvn** [comment]
Pulling https://github.com/istio/istio/pull/40449 or it's cherry-picks fixed the failing tests against 1.15 and 1.14.
